server/src/com/cloud/api/query/dao/SecurityGroupJoinDaoImpl.java

```diff
@@ -125,6 +130,14 @@ public SecurityGroupResponse newSecurityGroupResponse(SecurityGroupJoinVO vsg, A
             }
         }
 
+        List<SecurityGroupVMMapVO> securityGroupVmMap = _securityGroupVMMapDao.listBySecurityGroup(vsg.getId());
+        s_logger.debug("newSecurityGroupResponse() -> virtualmachine count: " + securityGroupVmMap.size());
+        sgResponse.setVirtualMachineCount(securityGroupVmMap.size());
+
+        for(SecurityGroupVMMapVO securityGroupVMMapVO : securityGroupVmMap) {
+            sgResponse.addVirtualMachineId(_userVmDao.findById(securityGroupVMMapVO.getInstanceId()).getUuid());
```

Comment:
Add a check that findById is not returning null. There is a possibility that VM may get deleted after the call to listBySecurityGroup() but before that map entry is processed in the for-loop.